### ISSUE TITLE
ytpl throwing an error on playlists with more than 100 items

### ISSUE THREAD

**Issue by barklebork** [TimeForANinja/node-ytpl#50]
Since about 30 minutes, ytpl fails to retrieve any playlist, even perfectly working ones used today  
Here are the logs :

```
TypeError: Cannot read property 'indexOf' of undefined
    at Object.exports.between (D:\Projets\discord-jukebox\node_modules\ytpl\lib\util.js:148:18)
    at IncomingMessage.<anonymous> (D:\Projets\discord-jukebox\node_modules\ytpl\lib\nonfirstpage.js:22:45)
    at IncomingMessage.emit (events.js:323:22)
    at endReadableNT (_stream_readable.js:1204:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

After debugging a little bit, seems like `parsedString` in `nonfirstpage.js` only contains `{ restart: 'now' }`.
Did YouTube release an update breaking the lib ?

**Comment by HasukoP** [comment]
I'm having the same issue.
I don't think it's YouTube-sided, as some playlists (one with 18 videos, and another with 50 videos) still work for me. But yeah, when it comes to over a hundred the error happens.

**Comment by barklebork** [comment]
Thanks @HasukoP I didn't notice I forgot to talk about the limit option  
Seems like the trick used to change between playlist pages is having issues

As @HasukoP said, ytpl still works with the default settings (limit = 100) but fails when I try to raise the limit 

The issue is easily reproducible on [RunKit](https://npm.runkit.com/ytpl) :
```javascript
await ytpl('https://www.youtube.com/playlist?list=PLFWwmC5ICgobfW2uqRpENgtda-HCjVoOw', {limit: 0})
```

**Comment by LamkasDev** [comment]
i'm having the same issue

**Comment by TimeForANinja** [comment]
...seems to be a problem related to cookies YouTube expects...
had it working for a sec but cleaning up my code broke it...

**Comment by TimeForANinja** [comment]
actually changing nothing about the code and requesting the same playlist 5 times fails 2 times...
feel like whatever youtube is trying to do is not yet distributed to all their servers

**Comment by barklebork** [comment]
> 
> 
> actually changing nothing about the code and requesting the same playlist 5 times fails 2 times...
> feel like whatever youtube is trying to do is not yet distributed to all their servers

Weird, seems like servers in France/Germany all have this problem: out of 10 requests I got 10 fails

**Comment by Murena7** [comment]
i'm having the same issue =(((

**Comment by HasukoP** [comment]
There's that module https://www.npmjs.com/package/youtube-playlist-info which works perfectly well to get playlists above 100 videos, but it's a kinda old one.
Maybe you could see, idk

**Comment by TimeForANinja** [comment]
> There's that module https://www.npmjs.com/package/youtube-playlist-info which works perfectly well to get playlists above 100 videos, but it's a kinda old one.
> Maybe you could see, idk

it uses the official api and needs an api-key
so can't look into that

**Comment by TimeForANinja** [comment]
would be cool if you guys could check out the branch #54 - worked for me but i'd be happy with a few more confirmations

you can use `npm i git://github.com/Roki100/node-ytpl#patch-1` to install it and the regular `require('ytpl')` to use it

**Comment by Murena7** [comment]
> would be cool if you guys could check out the branch #54 - worked for me but i'd be happy with a few more confirmations
> 
> you can use `npm i git://github.com/Roki100/node-ytpl#patch-1` to install it and the regular `require('ytpl')` to use it

This also worked for me !

**Comment by barklebork** [comment]
> 
> 
> would be cool if you guys could check out the branch #54 - worked for me but i'd be happy with a few more confirmations
> 
> you can use `npm i git://github.com/Roki100/node-ytpl#patch-1` to install it and the regular `require('ytpl')` to use it

Works for me too !

**Comment by TimeForANinja** [comment]
closed with #54 
gonna push to npm in a sec
